task6/part1.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+)
+
+type Info struct {
+	Host       string `json:"host"`
+	UserAgent  string `json:"user_agent"`
+	RequestURI string `json:"request_uri"`
+	Header     Header `json:"headers"`
+}
+
+type Header struct {
+	Accept    []string `json:"Accept"`
+	UserAgent []string `json:"User-Agent"`
+}
+
+func handler(w http.ResponseWriter, r *http.Request) {
+	info := Info{
+		Host:       r.Host,
+		UserAgent:  r.UserAgent(),
+		RequestURI: r.RequestURI,
+		Header: Header{
```

Comment:
You forget about about other headers that can be passed.
The HW task did not say that only this headers were needed =)
https://en.wikipedia.org/wiki/List_of_HTTP_header_fields
Example:
`curl "http://127.0.0.1:8081/headers" -H "Max-Forwards: 10"` should return:
```json
{
  "Host": "127.0.0.1:8080",
  "UserAgent": "curl/7.68.0",
  "RequestURI": "/headers",
  "Header": {
    "Accept": [
      "*/*"
    ],
    "Max-Forwards": [
      "10"
    ],
    "User-Agent": [
      "curl/7.68.0"
    ]
  }
}
```